pkg/util/test/util.go

```diff
@@ -643,6 +655,102 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validatePodArgs(cluster *corev1alpha1.StorageCluster, pod v1.Pod) error {
+	args := pod.Spec.Containers[0].Args
+	for i := 0; i < len(args); i++ {
+		switch args[i] {
+		case "-d":
+			if cluster.Spec.Network != nil && cluster.Spec.Network.DataInterface != nil {
+				if *cluster.Spec.Network.DataInterface != args[i+1] {
+					return fmt.Errorf("data network interface does not match. expected: %s, got: %s",
+						*cluster.Spec.Network.DataInterface, args[i+1])
+				}
+			} else {
+				return fmt.Errorf("data network interface expected to be set")
+			}
+			i++
+		case "-s":
```

Comment:
We need to check against cloud drive specs as well. 
`-s` is used for both normal storage as well as cloud drive storage. So it could have a value from either `spec.storage.devices` or `spec.cloudStorage.deviceSpecs`.
Same logic applies to `-j` and `-metadata`

Also check these storage args -
spec.cloudStorage.MaxStorageNodes (-max_drive_set_count)
spec.cloudStorage.MaxStorageNodesPerZone (-max_storage_nodes_per_zone)